datafusion/optimizer/src/pre_cast_lit_in_comparison.rs

```diff
@@ -318,6 +317,20 @@ mod tests {
         assert_eq!(optimize_test(expr_lt, &schema), expected);
     }
 
+    #[test]
+    fn nested() {
```

Comment:
Does this test require this pr to pass?
I think the original logic can handle this case

Reply:
No, this is just an additional test to make sure that this PR did not cause any regressions.